packages/orm/src/collections/OneToOneReference.ts

```diff
@@ -35,6 +48,7 @@ export class OneToOneReference<T extends Entity, U extends Entity>
     public otherFieldName: keyof U,
   ) {
     super();
+    this.otherColumnName = otherMeta.columns.find((field) => field.fieldName === otherFieldName)!.columnName;
```

Comment:
@stephenh is this going to have load order issues? Hoping not to have to make codegen changes.

Reply:
We should be good. The `metadata` structure should be fully initialized before any entity instances are created.